pkg/models/resources/pods.go

```diff
@@ -30,16 +31,155 @@ import (
 type podSearcher struct {
 }
 
+func (*podSearcher) get(namespace, name string) (interface{}, error) {
+	return informers.SharedInformerFactory().Core().V1().Pods().Lister().Pods(namespace).Get(name)
+}
+
+func podBelongTo(item *v1.Pod, kind string, name string) bool {
+	switch kind {
+	case "Deployment":
+		if podBelongToDeployment(item, name) {
+			return true
+		}
+	case "ReplicaSet":
+		if podBelongToReplicaSet(item, name) {
+			return true
+		}
+	case "DaemonSet":
+		if podBelongToDaemonSet(item, name) {
+			return true
+		}
+	case "StatefulSet":
+		if podBelongToStatefulSet(item, name) {
+			return true
+		}
+	case "Job":
+		if podBelongToJob(item, name) {
+			return true
+		}
+	}
+	return false
+}
+
+func replicaSetBelongToDeployment(replicaSet *v12.ReplicaSet, deploymentName string) bool {
+	for _, owner := range replicaSet.OwnerReferences {
+		if owner.Kind == "Deployment" && owner.Name == deploymentName {
+			return true
+		}
+	}
+	return false
+}
+
+func podBelongToDaemonSet(item *v1.Pod, name string) bool {
+	for _, owner := range item.OwnerReferences {
+		if owner.Kind == "DaemonSet" && owner.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+func podBelongToJob(item *v1.Pod, name string) bool {
+	for _, owner := range item.OwnerReferences {
+		if owner.Kind == "Job" && owner.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+func podBelongToReplicaSet(item *v1.Pod, replicaSetName string) bool {
+	for _, owner := range item.OwnerReferences {
+		if owner.Kind == "ReplicaSet" && owner.Name == replicaSetName {
+			return true
+		}
+	}
+	return false
+}
+
+func podBelongToStatefulSet(item *v1.Pod, statefulSetName string) bool {
+	for _, owner := range item.OwnerReferences {
+		if owner.Kind == "StatefulSet" && owner.Name == statefulSetName {
+			return true
+		}
+	}
+	return false
+}
+
+func podBelongToDeployment(item *v1.Pod, deploymentName string) bool {
+	replicas, err := informers.SharedInformerFactory().Apps().V1().ReplicaSets().Lister().ReplicaSets(item.Namespace).List(labels.Everything())
+	if err != nil {
+		return false
+	}
+
+	for _, r := range replicas {
+		if replicaSetBelongToDeployment(r, deploymentName) && podBelongToReplicaSet(item, r.Name) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func podBindPVC(item *v1.Pod, pvcName string) bool {
+	for _, v := range item.Spec.Volumes {
+		if v.VolumeSource.PersistentVolumeClaim != nil &&
+			v.VolumeSource.PersistentVolumeClaim.ClaimName == pvcName {
+			return true
+		}
+	}
+	return false
+}
+
+func podBelongToService(item *v1.Pod, serviceName string) bool {
+	service, err := informers.SharedInformerFactory().Core().V1().Services().Lister().Services(item.Namespace).Get(serviceName)
+	if err != nil {
+		return false
+	}
+	for k, v := range service.Spec.Selector {
```

Comment:
client-go has a more elegant way to determine if service selects a pod
 https://github.com/kubernetes/client-go/blob/89226ff90625425e6c7a87fe219dcefd75d28e53/listers/core/v1/service_expansion.go#L36
```
		selector := labels.Set(service.Spec.Selector).AsSelectorPreValidated()
		if selector.Matches(labels.Set(pod.Labels)) {
			services = append(services, service) // means service selected this pod
		}
```